CovertInfiltration/Src/CovertInfiltration/Classes/X2EventListener_Infiltration_UI.uc

```diff
@@ -510,19 +513,50 @@ static function EventListenerReturn UpdateResources(Object EventData, Object Eve
 	{
 		if (
 			CurrentScreen == CovertActions ||
-			(CovertActions.SSManager != none && CovertActions.SSManager.ShouldShowResourceBar() && CurrentScreen.IsA(class'UISquadSelect'.Name))
-		) {
+			(CovertActions.SSManager != none && CovertActions.SSManager.ShouldShowResourceBar() && CurrentScreen.IsA(class'UISquadSelect'.Name)))
+		{
+			CurrentBarracksStatus = class'X2Helper_Infiltration'.static.GetBarracksStatusReport();
 			// Just do same thing as done for UICovertActions
 			AvengerHUD.UpdateSupplies();
 			AvengerHUD.UpdateIntel();
 			AvengerHUD.UpdateAlienAlloys();
 			AvengerHUD.UpdateEleriumCrystals();
 
+			if (CurrentBarracksStatus.Tired > 0)
+			{
+				AvengerHUD.AddResource(default.strTiredSoldiers, class'UIUtilities_Text'.static.GetColoredText(string(CurrentBarracksStatus.Tired), eUIState_Warning));
+			}
+			AvengerHUD.AddResource(default.strReadySoldiers, class'UIUtilities_Text'.static.GetColoredText(string(CurrentBarracksStatus.Ready), eUIState_Good));
+
 			// Resource bar is hidden by default, show it
 			AvengerHUD.ShowResources();
 		}
 	}
 
+	////////////////////////
+	/// UIMission screen ///
+	////////////////////////
+
+	if (UIMission(CurrentScreen) != none)
```

Comment:
also check that it's **not** `UIMission_Infiltrated` (since that infiltration already in progress, no need to show soldier counts)